spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java

```diff
@@ -1473,9 +1503,371 @@ static void queryWithQueryOptions(DatabaseClient dbClient) {
+  }
+  // [END spanner_list_database_operations]
+
+  // [START spanner_list_backups]
+  static void listBackups(InstanceAdminClient instanceAdminClient, DatabaseId database) {
+    Instance instance = instanceAdminClient.getInstance(database.getInstanceId().getInstance());
+    // List all backups.
+    System.out.println("All backups:");
+    for (Backup backup : instance.listBackups().iterateAll()) {
+      System.out.println(backup);
+    }
+
+    // List all backups whose name contains 'users'.
+    System.out.println("All backups with backup name containing \"users\":");
+    for (Backup backup : instance.listBackups(Options.filter("name:users")).iterateAll()) {
+      System.out.println(backup);
+    }
+
+    // List all backups for databases whose name contains the text 'bank'.
+    System.out.println("All backups for databases with a name containing \"bank\":");
+    for (Backup backup : instance.listBackups(Options.filter("database:bank")).iterateAll()) {
+      System.out.println(backup);
+    }
+
+    // List all backups that expire before a certain time.
+    System.out.println("All backups that expire before 2019-10-18T02:56:53Z:");
+    for (Backup backup :
+        instance
+            .listBackups(Options.filter("expire_time < \"2019-10-18T02:56:53Z\""))
+            .iterateAll()) {
+      System.out.println(backup);
+    }
+
+    // List all backups with size greater than a certain number of bytes.
+    System.out.println("All backups with size greater than 1000 bytes:");
+    for (Backup backup : instance.listBackups(Options.filter("size_bytes > 1000")).iterateAll()) {
+      System.out.println(backup);
+    }
+
+    // List all backups with a create time after a certain timestamp and that are also ready.
+    System.out.println("All databases created after 2019-10-18T02:56:53Z and that are ready:");
+    for (Backup backup :
+        instance
+            .listBackups(Options.filter("create_time >= \"2019-10-18T02:56:53Z\" AND state:READY"))
+            .iterateAll()) {
+      System.out.println(backup);
+    }
+
+    // List backups using pagination.
+    Page<Backup> page = instance.listBackups(Options.pageSize(10));
+    while (true) {
+      for (Backup backup : page.getValues()) {
+        System.out.println(backup);
+      }
+      if (!page.hasNextPage()) {
+        break;
+      }
+      page = page.getNextPage();
+    }
+  }
+  // [END spanner_list_backups]
+
+  // [START spanner_restore_backup]
+  static void restoreBackup(
+      DatabaseAdminClient dbAdminClient,
+      DatabaseId sourceDatabaseId,
+      DatabaseId restoreToDatabase) {
+    // Get the most recent sample backup of the sample database.
+    String backupName =
+        String.format(
+            "%s_%02d",
+            sourceDatabaseId.getDatabase(), LocalDate.now().get(ChronoField.ALIGNED_WEEK_OF_YEAR));
+    BackupId backupId = BackupId.of(sourceDatabaseId.getInstanceId(), backupName);
```

Comment:
We ended up passing in the backupId as an argument in the other languages. Could we also do that here?

Reply:
done